### ISSUE TITLE
Gnome 45 Testing

### ISSUE THREAD

**Issue by ZachBacon** [getsolus/packages#617]
# GNOME 45 Stack Upgrade Notes

WARNING: Please read this task in its entirety before commenting or upgrading. Seriously.

As is tradition, I'm opening up a task for everyone to report their issues with the GNOME 45 stack upgrade. It's absolutely important we maintain focus on reporting issues which are specifically related to this stack upgrade so we can best work to helping validate and resolve these issues. I welcome you to post a comment if an application is suddenly not working and you strongly believe it is the result of the new stack upgrade, such as it suddenly missing libraries or ones which may warrant a rebuild. If necessary, we may request you file a separate issue so we may obtain further detailed information, which we can reference in this task.

NOTE: This task content can only be modified by members of the Core Team or Global Maintainers. We will update the task as comments / feedback are received.

The objective of this task is to facilitate **technical** discussions around changes which have occurred in the GNOME Stack, such as:

- Changes to GNOME Shell 45
- GNOME Applications

This testing is being performed on the unstable repository and may also involve or require real-time communication via our development matrix room. If you are not in matrix, use any supported Matrix client that we provide in the repo (e.g. Element, Nheko, Neochat) and look at [our Getting Involved Page](https://help.getsol.us/docs/user/contributing/getting-involved#matrix-chat) for connection details.

If you are not on the unstable repository and wish to help test, assuming you understand that the intent of unstable is to be the place where we break stuff and operate under the assumption you check the topic of the Packaging Matrix room to know **when not to update**, you can run the below mentioned command to switch to the unstable repository:

```
sudo eopkg ar Solus https://cdn.getsol.us/repo/unstable/eopkg-index.xml.xz
```

## Upgrading Properly

**Do not run any commands until you fully read this section. Thank you.**

I cannot stress enough the importance of ensuring you upgrade properly and fully. If you are not on unstable already, you should **fully upgrade** your system by opting in to every package update available via the Software Center or running `sudo eopkg up` then reboot **before** switching over to unstable.

If you are on the unstable repository, likewise perform a **full upgrade** via either the Software Center or the Terminal, then reboot. If you use Budgie and GNOME and encounter the grey "Something happened" fullscreen window after a reboot and attempt to login, switch to a TTY (typically Ctrl+Alt+F2-9, pick one), login and run `sudo usysconf run -f` then reboot (`sudo reboot`).

If you are using GNOME Shell, you **need** to run the following command after the upgrade if you use GNOME Shell to properly reset your extensions back to the stock configuration:

```
gsettings reset org.gnome.shell 'enabled-extensions'
```

Seriously, be sure to **disable any extensions** which we do not provide / have enabled out-of-the-box. Otherwise you may have a completely broken GNOME Shell experience, or worse it'll crash and you'll have to go through loads of hoops such as hand modifying dconf key values and nuking some extension contents, etc. It is not my responsibility nor the responsibility of GNOME to ensure those extensions are maintained and updated against newer GNOME Shell APIs. If they break, you'll need to talk to the extension developer. Additionally ensure you are not updating the extensions we support and directly provide, such as Tray Icons Reloaded, through any software such as Extensions app and website It may result in a broken installation. If you "upgraded" extensions through that manner, it's suggested you uninstall them and reinstall the package equivalent if possible.

WARNING: Like I said earlier, reboot your system after upgrading. For real.

## GNOME Shell

GNOME Shell 45 predictably has a fundamentally changed user experience. I am shipping a slightly less modified GNOME Shell with this stack upgrade and some default theme changes for it.

1. The remaining extensions enabled by default, like Tray-Icons-Reloaded, are still supported and enabled.

### GNOME Shell

As with any desktop environment, there are wide range of issues with GNOME Shell. I would advise checking https://gitlab.gnome.org/GNOME/gnome-shell/issues before reporting them here. If there is a patch and we don't have it yet, I'll see about getting it added.

### Mutter

I currently have a couple patches in Mutter allow for triple buffering, should improve the experience. But it's an experimental patch that's not yet [upstream](https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/1441)

## New Applications

 - Gnome-Text-Editor
 - Gnome-Console

## Testing List

**Testing Notes:**

1. This list may not be comprehensive. Feel free to report other things!
2. github may indicate something is complete even if it isn't explicitly checked off. Ignore that and only look for the checkbox.

- [x] Bluetooth
  - Tested with Bluetooth pairing in GNOME Control Center with phone.
- [ ]  Desktops
  - [ ]  Budgie Desktop
     - [ ] GDM (when co-installed with GNOME Shell)
     - [x] LightDM + slick-greeter
  - [x] GNOME Shell
    - [x] GDM
  - [x] MATE
- [x] Geolocation
  - Tested with GNOME Weather
- [x] IBUS
- [ ] Networking
  - [ ] Shares (Samba via Nautilus)
  - [x] Wireless
  - [x] Wired
- [ ] Applications
  - [ ] Abiword
  - [x] Aisleriot
  - [x] Baobab (Disk Usage Analyzer)
  - [ ] Claws Mail
  - [x] Evince
  - [x] Evolution
  - [x] Four-in-a-Row
  - [x] Gaupol
  - [x] GEdit
  - [x] GIMP
  - [x] GNOME 2048
  - [x] GNOME Calculator
  - [x] GNOME Calendar
  - [x] Gnome Console
  - [x] Gnome Shell features
    - [x] Screencast recording selection
    - [x] Screencast recording whole screen
    - [x] Take a screenshot selection
    - [x] Take a screenshot window
    - [x] Take a screenshot wholescreen
    - [x] Night Light
    - [x] Dark theme toggle
  - [ ] GNOME Control Center
    - [x] About
    - [x] Background
    - [x] Bluetooth
    - [x] Date & Time
    - [ ] Devices
        - [x] Color
        - [x] Displays
        - [x] Keyboard
        - [ ] Mouse & Touchpad
        - [x] Printers
        - [x] Removable Media
    - [x] Networking
        - [x] Wi-Fi
        - [x] Wired
    - [x] Notifications
    - [x] Online Accounts (Tested with a Google Account)
    - [x] Power
    - [x] Privacy
    - [x] Region & Language
    - [x] Search (only applies to GNOME Shell)
    - [ ] Sharing
    - [x] Sound
    - [ ] Accessability
        - [ ] Seeing
        - [ ] Hearing
        - [ ] Typing
        - [ ] Pointing and Clicking
        - [ ] Zoom
  - [x] GNOME Disks
  - [x] GNOME Help / Yelp
  - [x] GNOME Nibbles
  - [x] GNOME Mines
  - [x] Eye of GNOME
  - [x] GNOME Pomodoro
  - [x] GNOME Power Statistics (GNOME Power Manager)
  - [x] GNOME Logs
  - [x] Celluloid
  - [x] GNOME Mahjongg
  - [x] GNOME Maps
  - [x] GNOME Music
  - [x] GNOME Sudoku
  - [x] Gnome-Text-Editor
  - [x] GNOME Tweaks
  - [x] GNOME Videos (Totem)
  - [x] Glade
  - [x] Gthumb
  - [x] Lollypop
  - [x] Lyx
  - [x] Nautilus
  - [x] Pitivi
  - [x] Polari
  - [x] Quodlibet
  - [x] Rhythmbox
  - [ ] Seahorse
  - [x] Simple Scan
  - [x] Shotwell
  - [x] Solus Software Center
  - [x] Transmission
  - [ ] Vocal
  - [x] budgie-extras applets if you happen to use them
  - [x] gtkhash (optionally the caja extension)

**Comment by joebonrichie** [comment]
i'm not sure if gnome-browser-connector is packaged properly

> This repository contains OS-native connector counterpart for GNOME Shell browser extension.

currently `gnome-browser-connector` replaces `chrome-gnome-shell`. However, `gnome-browser-extension` should be packaged and replace `chrome-gnome-shell` instead?

yeah https://gitlab.gnome.org/GNOME/chrome-gnome-shell redirects to gnome-browser-extension

**Comment by ZachBacon** [comment]
That can be rectified soon.

**Comment by EbonJaeger** [comment]
Budgie rebuilds are in #619

**Comment by EbonJaeger** [comment]
**Budgie Users:** Note that after updating, the Power Dialog to log out, reboot, shutdown, etc. will not work. The journal logs say something about the pixmap provider for icons not being able to load an icon, possibly as a result of the glibc toolchain update. Not sure yet. As such, you will have to use alternate means to reboot or shut down your computer. The problem seems to go away after rebooting.

**Comment by EbonJaeger** [comment]
MATE also has an issue when trying to shut down right after updating. Found this in the journal logs: https://bpa.st/GIDA

**Comment by Staudey** [comment]
> Updates mutter so that it's split into mutter and mutter common so that budgie can make use of the common package for their window manager

This had already been done for GNOME 44.3. I would choose a different example or remove it to avoid confusion.

**Comment by EbonJaeger** [comment]
Edited the task to remove confusing list item.

**Comment by malfisya** [comment]
`evince` and `gnome-weather` works fine

**Comment by EbonJaeger** [comment]
Checked those off the list. Thanks!

**Comment by malfisya** [comment]
**Works fine** :
- [x] aisleriot 
- [x] gaupol 
- [x] gimp
- [x] gnome-power-manager 
- [x] gnome-mahjongg 
- [x] lyx 

**Problem** :
- [ ] pitivi 

> ERROR - The following hard dependencies are unmet:
> ==================================================
> - matplotlib not found on the system
> Missing soft dependency:
> - cvtracker GStreamer element not found on the system
>     -> enables object tracking
> Missing soft dependency:
> - librosa not found on the system
>     -> enables beat detection functionality

- [ ] transmission freeze
- [ ] vocal freeze a lot

**Comment by malfisya** [comment]
Update :
- `lollypop` and `quodlibet` works fine
- `transmission` works fine if you agree to EULA first
- `vocal` still freezes a lot
-  `pitivi` still same error (maybe i will try submit a PR if if figure this out) :crossed_fingers:

**Comment by malfisya** [comment]
**Works fine**:
[x] shotwell
[x] totem
[x] evolution

**Not Working**:
[ ] abiword | icon not showing on overview
[ ] gnome-music
```
(org.gnome.Music:37559): libsoup-ERROR **: 18:23:51.684: libsoup3 symbol detected. Using libsoup2 and libsoup3 in the same process is not supported.
Trace/breakpoint trap (core dumped)
```

**Comment by EbonJaeger** [comment]
GNOME Music should work now.

**Comment by joebonrichie** [comment]
I'm experiencing segfaults again using the screen recording feature. Wayland Nvidia.

```
Stack trace of thread 928297:
#0  0x00007f576f6024d9 gst_is_dmabuf_memory (libgstallocators-1.0.so.0 + 0x24d9)
#1  0x00007f576ecc52d9 _dma_buf_upload_accept (libgstgl-1.0.so.0 + 0x3d2d9)
#2  0x00007f576ecc631c gst_gl_upload_perform_with_buffer (libgstgl-1.0.so.0 + 0x3e31c)
#3  0x00007f576dbc1b9c gst_gl_upload_element_prepare_output_buffer (libgstopengl.so + 0xdb9c)
#4  0x00007f576ec4272b default_generate_output (libgstbase-1.0.so.0 + 0x4972b)
#5  0x00007f576ec420a6 gst_base_transform_chain (libgstbase-1.0.so.0 + 0x490a6)
#6  0x00007f576ff626e7 gst_pad_chain_data_unchecked (libgstreamer-1.0.so.0 + 0x7b6e7)
#7  0x00007f576ff64ee6 gst_pad_push_data (libgstreamer-1.0.so.0 + 0x7dee6)
#8  0x00007f576ff6b364 gst_pad_push (libgstreamer-1.0.so.0 + 0x84364)
#9  0x00007f576ec42183 gst_base_transform_chain (libgstbase-1.0.so.0 + 0x49183)
#10 0x00007f576ff626e7 gst_pad_chain_data_unchecked (libgstreamer-1.0.so.0 + 0x7b6e7)
#11 0x00007f576ff64ee6 gst_pad_push_data (libgstreamer-1.0.so.0 + 0x7dee6)
#12 0x00007f576ff6b364 gst_pad_push (libgstreamer-1.0.so.0 + 0x84364)
#13 0x00007f576ec3defd gst_base_src_loop (libgstbase-1.0.so.0 + 0x44efd)
#14 0x00007f576ff95073 gst_task_func (libgstreamer-1.0.so.0 + 0xae073)
#15 0x00007f579651e8d3 g_thread_pool_thread_proxy (libglib-2.0.so.0 + 0x898d3)
#16 0x00007f579651df18 g_thread_proxy (libglib-2.0.so.0 + 0x88f18)
#17 0x00007f57960f410a start_thread (libc.so.6 + 0x9710a)
#18 0x00007f5796181cbc __clone3 (libc.so.6 + 0x124cbc)
```

**Comment by malfisya** [comment]
> GNOME Music should work now.

Confirmed works. Thank you

**Comment by TraceyC77** [comment]
Reported on [the forums](https://discuss.getsol.us/d/9904-sync-update-for-weeks-42-and-43-2023/31):

> Hey Folks, I just ran into a GNOME X11 issue: sometimes, and that for basically as long as I can remember, GNOME app switching shortcut (super+tab) behaves strange. This time it was super+tab didn't switch the app, but only brought the app switching dialog/window, I then had to keep pressing super and repress tab to actually switch the app. Behaviours like that I normally resolve by restarting the GNOME shell with Alt+F2, r, Enter.
> Doing so, the shell apparently crashes: I get a white screen with error message: An error occured that could't be resolved by the system. User has to log out. (roughly translated from the german error message). I then can only click log user out and get logged out to gdm login screen, from where I can log in back, but all running windows then are closed, of course.
> I reproduced this now 3 times in a row. As long as there are open windows (in my case nheko and GNOME calendar) and you perform the GNOME shell restart with Alt+F2, r, Enter, the crash occurs.

**Comment by joebonrichie** [comment]
I can replicate the issue where restarting the gnome shell crashes the session on X11

**Comment by sebo505** [comment]
The crash of the session when restarting the GNOME shell under X11 is an upstream issue. [Here](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/7050) is a link to the thread on the GNOME gitlab issue tracker.

**Comment by joebonrichie** [comment]
Just backported a patch to fix restarting gnome shell on X11: e7c2d21bacefd9ea30c60dc2e9aa3be980cf74ce

**Comment by malfisya** [comment]
- pitivi works again
- polari works
- simple-scan works
- bluetooth with headset and phone are tested
- wi-fi works
- wired works
- printer works
- removable media works
- dark theme toggle works
